programs/ssl/ssl_client2.c

```diff
@@ -2239,7 +2240,10 @@ int main( int argc, char *argv[] )
                     "    or you didn't set ca_file or ca_path "
                         "to an appropriate value.\n"
                     "    Alternatively, you may want to use "
-                        "auth_mode=optional for testing purposes.\n" );
+                        "auth_mode=optional for testing purposes if "
+                        "server is not TLS 1.3.\n"
```

Comment:
It would be nice to check that auth_mode=optional is not used with TLS 1.3 and error if this is the case.

Also - better to say e.g. "if not using TLS 1.3" rather than "if server is not TLS 1.3" as it's not specifically relating to the server but to the use of TLS 1.3 overall?

Reply:
> It would be nice to check that auth_mode=optional is not used with TLS 1.3 and error if this is the case.

That depends on the negotiate version. Difficult to check it in configuration stage.  I just add config check to see if CI happy.